### PR TITLE
Fix #4012 AWS Java SDK libraries outdated

### LINKED ISSUE #4012: AWS Java SDK libraries are too old
Trying to publish to an s3 bucket in Canada, but get:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':publishMavenJavaPublicationToMaven2Repository'.
> Failed to publish publication 'mavenJava' to repository 'maven2'
   > Could not write to resource 's3://redacted'.
      > Not able to determine region for ca-central-1.Please upgrade to a newer version of the SDK
```
This is because the `aws-java-sdk-core-1.11.6.jar` is missing the Canada region that was introduced back in 2016.

Can these libraries be updated?

### PR DESCRIPTION
**Pull request by dhoer**
### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/.github/CONTRIBUTING.md)
- [x] Make sure that all commmits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/resources-s3/src/main/java/org/gradle/internal/resource/transport/aws/s3/S3Client.java [hunk 1]
```diff
@@ -89,7 +90,7 @@ private void setAmazonS3ConnectionEndpoint() {
         Optional<URI> endpoint = s3ConnectionProperties.getEndpoint();
         if (endpoint.isPresent()) {
             amazonS3Client.setEndpoint(endpoint.get().toString());
-            clientOptionsBuilder.setPathStyleAccess(true);
+            clientOptionsBuilder.setPathStyleAccess(true).disableChunkedEncoding();
```
Comment (blindpirate) [inline subprojects/resources-s3/src/main/java/org/gradle/internal/resource/transport/aws/s3/S3Client.java]: Could you tell me why `disableChunkedEncoding()` is necessary here?

**Comment by dhoer** [comment]
@blindpirate After 1.11.15, Amazon client uses chunked encoding by default (rather than sending the raw bytes), and s3 ends up calculating the md5 on encoded payload, not the raw data.  

I will see if your change passes the tests. I had issues with AmazonS3 being immutable when I originally tried that.

**Comment by blindpirate** [comment]
Thanks, please let me know the result.

**Comment by dhoer** [comment]
@blindpirate Still getting  `java.lang.UnsupportedOperationException: Client is immutable when created with the builder.` error.  It would require a rewrite to get this fixed. @oehme Has scheduled a follow-up issue for 5.0 to refactor resource-s3 to use AWS Java SDK 2.0.  So I would recommend going with suppressed deprecation warnings for gradle 4.x.

**Comment by blindpirate** [comment]
> So I would recommend going with suppressed deprecation warnings for gradle 4.x.

Okay, that makes sense. However, this PR keeps failing on our CI, could you do the following two things or allow us to push commits to your branch?

- Update [this number](https://github.com/gradle/gradle/blob/29dcd9a47981b4ffc1b21353063be3c4fdcb38ef/subprojects/distributions/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy#L41) to `186`
- Rebase `master` to pick up latest changes on `master`. There're some performance issues on your branch and it blocks the tests.

**Comment by blindpirate** [comment]
Thanks @dhoer !

**Pull request merged**